components/controls/EscKeydownHandler.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Sonar UI Common
+ * Copyright (C) 2019-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import { KeyCodes } from '../../helpers/keycodes';
+
+interface Props {
+  children: React.ReactNode;
+  onKeydown: () => void;
+}
+
+export default class EscKeydownHandler extends React.Component<Props> {
+  componentDidMount() {
+    setTimeout(() => {
+      document.addEventListener('keydown', this.handleKeyDown, false);
+    }, 0);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.handleKeyDown, false);
+  }
+
+  handleKeyDown = (event: KeyboardEvent) => {
+    if (event.keyCode === KeyCodes.Escape) {
```

Comment:
`keyCode` is [deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode), seems like the replacement is `key` or `code`. Maybe other keydown handler should be updated too ?

Reply:
Yeah, went for the quick'n'easy way. The thing is, `code` isn't supported in IE11, and `key` is different ("Esc" rather than "Escape"). You think we should still use the new API? I can check for both cases.

Reply:
And not supported in Edge either no ?
I think we should use the new API sooner or later because we might end up with a recent browser dropping it, but I don't see any urgency to do it and indeed it does seem more complicated to use the new API. Up to you, I'm fine in both cases.

Reply:
FYI: we use `keyCode` in many places, but through the `React.KeyboardEvent` interface, where it is not deprecated. Here's your solution 🤣

Reply:
I'll keep it like this for now. And as we're using the `KeyCodes` enum, it will be easy to find this occurrence.